addons/binding/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/model/OshiSysteminfo.java

```diff
@@ -182,6 +193,9 @@ public DecimalType getMemoryUsed() {
 
     @Override
     public DecimalType getStorageTotal(int index) throws DeviceNotFoundException {
+        // In the current OSHI version a new query is required for the storage data values to be updated
+        // In OSHI 4.0.0. it is planed to change this mechanism - see https://github.com/oshi/oshi/issues/310
+        fileStores = operatingSystem.getFileSystem().getFileStores();
```

Comment:
Do we still need the field filestores or should we just get rid of it (for now)

Reply:
Given that we did not update the dependency does that mean the current version is broken? I feel we need this now?

Reply:
For some methods, such as getStorageName(), we assume the name isn't going to be changed, so we do not need to recreate the fileStores variable in that method.

Reply:
The current OSHI version works fine, but currently in some of the queried methods the data is not updated dynamically, so lines like this need to be called in such methods:

 `fileStores = operatingSystem.getFileSystem().getFileStores();`
The